Common/Source/Dialogs/dlgStatus.cpp

```diff
@@ -239,22 +239,17 @@ static void UpdateValuesSystem() {
     if (VarioAvailable(GPS_INFO)) {
 	// LKTOKEN  _@M199_ = "Connected"
 #ifdef DEVICE_SERIAL
-     if(GPS_INFO.HardwareId >0)
+     if(DeviceList[ GPS_INFO.VarioSourceIdx ].HardwareId >0)
```

Comment:
GPS_INFO and DeviceList is shared data, mutex lock is required.

Reply:
How, correctly?